### ISSUE TITLE
Build failure caused by multiple specialization on installed wybelib

### ISSUE THREAD

**Issue by CaviarChen** [pschachte/wybe#130]
By default, the wybelib is intalled in a folder that the normal users do not have write permission.
If the compiler tries to generate a specialized version for a procedure in stdlib, it fails.
Now, our stdlib is complex enough to have this problem.
Example:
```
$ cd test-cases/execution
$ wybemk echo
FdStreamException "Permission denied"
```

I think in this case, the compiler should copy the object file to a local folder, such as `./_build/`.

We should also consider to cover this in our CI by using the installed wybemk to run all tests.

**Comment by pschachte** [comment]
This is a really tricky problem.  It doesn't sound ideal to have the compiler make local copies of whole library modules.  As the library grows, and particularly when only a few procs in many large library modules need specialisation, this could get expensive.  What about generating a single local file of specialised versions of any library (or generally non-overwritable object) code?  This would be an exception to the one-one relationship between object files and source files.  But I actually already broke that a little in fixing support for module directories (I generate a separate object file for whatever code and metaadata is contained in the directory module).

**Comment by CaviarChen** [comment]
I personally don't think this is that tricky.
It's all about granularity, we can make a copy per project (like I mentioned) or per user (put it in somewhere like `~/.wybe/cache`). We can have a separate file for specialized versions or even for each procedure.
Switching between approches is also not that difficult. I recommend we just take whatever that is reasonable and easy to implement.
Mainly I am kind of against optimizing (and spend time on it) too early, we can always improve it when it become a real problem.
Based on sizes current wybe programs, performance of cpu (also we do support incremental build) and cost of disk space, the compiler architecture could be a lot different when that happens.

**Comment by pschachte** [comment]
Fixed in commit #139
